pkg/cluster/deploybaseresources.go

```diff
@@ -201,6 +175,39 @@ func (m *manager) deployBaseResourceTemplate(ctx context.Context) error {
 	return arm.DeployTemplate(ctx, m.log, m.deployments, resourceGroup, "storage", t, nil)
 }
 
+func (m *manager) newPublicLoadBalancer(ctx context.Context, resources *[]*arm.Resource) {
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+	azureRegion := strings.ToLower(m.doc.OpenShiftCluster.Location) // Used in k8s object names, so must pass DNS-1123 validation
+
+	var outboundIPs []api.ResourceReference
+	if m.doc.OpenShiftCluster.Properties.APIServerProfile.Visibility == api.VisibilityPublic {
+		*resources = append(*resources,
+			m.networkPublicIPAddress(azureRegion, infraID+"-pip-v4"),
+		)
+		if m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs != nil {
+			outboundIPs = append(outboundIPs, api.ResourceReference{ID: m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID + "/providers/Microsoft.Network/publicIPAddresses/" + infraID + "-pip-v4"})
+		}
+	}
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs != nil {
+		for i := len(outboundIPs); i < m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count; i++ {
+			ipName := genManagedOutboundIPName()
+			*resources = append(*resources, m.networkPublicIPAddress(azureRegion, ipName))
+			outboundIPs = append(outboundIPs, api.ResourceReference{ID: m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID + "/providers/Microsoft.Network/publicIPAddresses/" + ipName})
+		}
+	}
+	m.patchEffectiveOutboundIPs(ctx, outboundIPs)
+	// Normal private clusters still need a public load balancer
```

Comment:
I realize that this comment was just moved into a function and you didn't change it as part of this PR, but would you mind clarifying what it means either in this discussion or directly in the code (as appropriate)? It reads to me as if this code has been moved from a context where the comment made more sense.

Reply:
Agreed, I removed this comment.  I think it is clear in the code why we need a public load balancer (outbound routing).